File: lee/src/main/java/org/example/lee/题目/动态规划/b_最大子数组和.java
package org.example.lee.题目.动态规划;

public class b_最大子数组和 {

  public int maxSubArray(int[] nums) {
    int h = 0, max = Integer.MIN_VALUE;
    for (int k : nums) {
      h += k;
      max = Math.max(max, h);
      if (h < 0) {
        h = 0;
      }
    }
    return max;
  }

}

File: lee/src/main/java/org/example/lee/题目/动态规划/b_最大子数组和.go
package dp

func maxSubArray(nums []int) int {
	Max, h := nums[0], 0
	for i := 0; i < len(nums); i++ {
		h += nums[i]
		Max = max(Max, h)
		if h <= 0 {
			h = 0
		}
	}
	return Max
}
